src/jobManagerClient.ts

```diff
@@ -10,208 +10,186 @@ import {
+        version: findJobsParams.version,
+        isCleaned: findJobsParams.isCleaned,
+        productType: findJobsParams.productType,
+        status: findJobsParams.status,
+        type: findJobsParams.type,
+        shouldReturnTasks: findJobsParams.shouldReturnTasks,
+        fromDate: findJobsParams.fromDate,
+        tillDate: findJobsParams.tillDate,
+        internalId: findJobsParams.internalId,
+      });
       if (typeof res === 'string' || res.length === 0) {
         return [];
       }
       return res;
     } catch (err) {
-      this.logger.error(
-        `[JobManagerClient][getJobByInternalId] internalId=${internalId}, failed error=${JSON.stringify(err, Object.getOwnPropertyNames(err))}`
-      );
+      this.logger.error({ err, url: getJobsUrl, targetService: this.targetService, findJobsParams }, `failed to getJobs`);
       throw err;
     }
   }
 
-  public async consume<T>(): Promise<ITaskResponse<T> | null> {
+  public async consume<T>(taskType: string): Promise<ITaskResponse<T> | null> {
+    const consumeTaskUrl = `/tasks/${this.jobType}/${taskType}/startPending`;
     try {
-      const consumeTaskUrl = `/tasks/${this.jobType}/${this.taskType}/startPending`;
       const taskResponse = await this.post<ITaskResponse<T>>(consumeTaskUrl);
       return taskResponse;
     } catch (err) {
       if (err instanceof NotFoundError) {
         return null;
       } else {
         this.logger.error(
-          `[JobManagerClient][consume] jobType=${this.jobType}, taskType=${this.taskType}, failed to consume error=${JSON.stringify(
-            err,
-            Object.getOwnPropertyNames(err)
-          )}`
+          { err, url: consumeTaskUrl, targetService: this.targetService, jobType: this.jobType, taskType: taskType },
+          `failed to consume task`
         );
         throw err;
       }
     }
   }
 
   public async enqueueTask<T>(jobId: string, payload: ICreateTaskBody<T>): Promise<void> {
+    const createTaskUrl = `/jobs/${jobId}/tasks`;
     try {
-      this.logger.debug(`[JobManagerClient][enqueueTask] jobId=${jobId}, payload=${JSON.stringify(payload)}`);
-      const createTaskUrl = `/jobs/${jobId}/tasks`;
+      this.logger.debug({ jobId, url: createTaskUrl, targetService: this.targetService, payload }, `enqueueTask for jobId=${jobId}`);
       await this.post(createTaskUrl, payload);
     } catch (err) {
-      this.logger.error(
-        `[JobManagerClient][enqueueTask] jobId=${jobId}, payload=${JSON.stringify(payload)} failed error=${JSON.stringify(
-          err,
-          Object.getOwnPropertyNames(err)
-        )}`
-      );
+      this.logger.error({ err, url: createTaskUrl, targetService: this.targetService, payload }, `failed to enqueueTask for jobId=${jobId}`);
       throw err;
     }
   }
 
   public async updateTask<T>(jobId: string, taskId: string, payload: IUpdateTaskBody<T>): Promise<void> {
+    const updateTaskUrl = `/jobs/${jobId}/tasks/${taskId}`;
```

Comment:
mabye creating private function to create url, `/jobs/${jobId}/tasks/${taskId}` accure very often

Reply:
fixed